### PR TITLE
Test getGroupID

### PR DESCRIPTION
**Pull request by klvimeo**

### CODE REVIEW

File: grouplogger.go [hunk 1]
```diff
@@ -259,22 +259,25 @@ func (gl *GroupLogger) getMaxSeverity() logging.Severity {
 	return max
 }
 
+func NewUUID() string {
```
Comment (justinruggles) [inline grouplogger.go]: Do you think this is generally useful to expose as a public function?

Comment (klvimeo) [reply grouplogger.go]: Nope, good catch.

File: grouplogger_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+const fake_uuid = "fake_uuid"
+
+func mockUUIDFunc() string {
+	return fake_uuid
+}
+
+func TestGetGroupIDWithRequestWithHeader(t *testing.T) {
+	r, _ := http.NewRequest("GET", "http://www.vimeo.com", nil)
+	r.Header.Set("X-Cloud-Trace-Context", "123")
+	id := getGroupID(r, newUUID)
```
Comment (justinruggles) [inline grouplogger_test.go]: use the mock one here

File: grouplogger.go [hunk 3]
```diff
@@ -74,7 +74,7 @@ func (client *Client) Logger(r *http.Request, name string, opts ...logging.Logge
 	// Use trace from request if available; otherwise generate a group ID.
 	gl := &GroupLogger{
 		Req:         r,
-		GroupID:     getGroupID(r),
+		GroupID:     getGroupID(r, newUUID),
```
Comment (justinruggles) [inline grouplogger.go]: just put the function inline here instead of defining it separately below

**Review by justinruggles — APPROVED** [review]

**Pull request merged**